### PR TITLE
Avoid creating ScalaInstance when not used

### PR DESCRIPTION
**Pull request by eed3si9n**

### CODE REVIEW

File: sbt-dotty/src/dotty/tools/sbtplugin/DottyPlugin.scala [hunk 1]
```diff
@@ -266,41 +269,39 @@ object DottyPlugin extends AutoPlugin {
...
-          // This should really be `old` with `val old = scalaInstance.value`
-          // above, except that this would force the original definition of the
-          // `scalaInstance` task to be computed when `isDotty` is true, which
-          // would fail because `managedScalaInstance` is false.
-          Defaults.scalaInstanceTask.value
-        }
+        val isD = isDotty.value
+        val si = scalaInstance.taskValue
+        val siTaskInitialize = Def.valueStrict { si }
+        if (isD) dottyScalaInstance
```
Comment (smarter) [inline sbt-dotty/src/dotty/tools/sbtplugin/DottyPlugin.scala]: Can we inline some of this stuff ? e.g.
````scala
if (isDotty.value) Def.task { ... }
else Def.valueStrict { ... }
```

Comment (eed3si9n) [reply sbt-dotty/src/dotty/tools/sbtplugin/DottyPlugin.scala]: I split things out so each step is easier to understand (for me). You can inline as much as you like.

**Comment by eed3si9n** [comment]
I am not sure if this is related to this change


```
[error] java.io.FileNotFoundException: http://repo1.maven.org/maven2/ch/epfl/lamp/dotty_0.16/maven-metadata.xml
[error] 	at sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1890)
[error] 	at sun.net.www.protocol.http.HttpURLConnection.access$200(HttpURLConnection.java:91)
[error] 	at sun.net.www.protocol.http.HttpURLConnection$9.run(HttpURLConnection.java:1484)
[error] 	at sun.net.www.protocol.http.HttpURLConnection$9.run(HttpURLConnection.java:1482)
[error] 	at java.security.AccessController.doPrivileged(Native Method)
[error] 	at java.security.AccessController.doPrivilegedWithCombiner(AccessController.java:782)
[error] 	at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1481)
[error] 	at java.net.URL.openStream(URL.java:1045)
[error] 	at scala.io.Source$.fromURL(Source.scala:140)
[error] 	at scala.io.Source$.fromURL(Source.scala:130)
[error] 	at dotty.tools.sbtplugin.DottyPlugin$autoImport$.dottyLatestNightlyBuild(DottyPlugin.scala:37)
[error] 	at ScalatestBuild$.dottyVersion$lzycompute(scalatest.scala:2136)
[error] 	at ScalatestBuild$.dottyVersion(scalatest.scala:2136)
[error] 	at ScalatestBuild$.$anonfun$dottySettings$1(scalatest.scala:2139)
[error] 	at sbt.internal.util.Init$Value.$anonfun$apply$3(Settings.scala:814)
[error] 	at sbt.internal.util.EvaluateSettings.$anonfun$constant$1(INode.scala:204)
[error] 	at sbt.internal.util.EvaluateSettings$MixedNode.evaluate0(INode.scala:221)
[error] 	at sbt.internal.util.EvaluateSettings$INode.evaluate(INode.scala:164)
[error] 	at sbt.internal.util.EvaluateSettings.$anonfun$submitEvaluate$1(INode.scala:87)
[error] 	at sbt.internal.util.EvaluateSettings.sbt$internal$util$EvaluateSettings$$run0(INode.scala:98)
[error] 	at sbt.internal.util.EvaluateSettings$$anon$3.run(INode.scala:94)
[error] 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[error] 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[error] 	at java.lang.Thread.run(Thread.java:748)
[error] java.io.FileNotFoundException: http://repo1.maven.org/maven2/ch/epfl/lamp/dotty_0.16/maven-metadata.xml
[error] Use 'last' for the full log.
Project loading failed: (r)etry, (q)uit, (l)ast, or (i)gnore?
```

File: sbt-dotty/src/dotty/tools/sbtplugin/DottyPlugin.scala (hunk 1 above)
Comment (eed3si9n) [reply sbt-dotty/src/dotty/tools/sbtplugin/DottyPlugin.scala]: c1aec9c82d0fde63918f700b35614baa18e688ca

**Comment by smarter** [comment]
Failure is unrelated, the dottyLatestNightlyBuild task from sbt-dotty is looking for a nightly of dotty_0.16 but we won't have one until tonight's build (because we just released 0.15 and bumped version today), someone should make that task more robust.

**Comment by anatoliykmetyuk** [comment]
The bug is fixed now, so this PR passes the CI.

**Pull request merged**